DeepSpeech.ipynb

```diff
@@ -1415,12 +1405,20 @@
    "source": [
     "# Define CPU as device on which the muti-gpu training is orchestrated\n",
     "with tf.device('/cpu:0'):\n",
-    "    # Obtain all the data, defaulting to TED LIUM\n",
-    "    data_sets = ds_importer_module.read_data_sets(tf.get_default_graph(), ds_dataset_path, batch_size, n_input, n_context)\n",
-    "    \n",
     "    # Create session in which to execute\n",
     "    session = tf.Session(config=tf.ConfigProto(allow_soft_placement=True, log_device_placement=True))\n",
     "    \n",
+    "    # Start queue runner threads\n",
+    "    tf.train.start_queue_runners(sess=session)\n",
+    "    \n",
+    "    # Obtain all the data, defaulting to TED LIUM\n",
+    "    data_sets = ds_importer_module.read_data_sets(session, ds_dataset_path, batch_size, n_input, n_context)\n",
+    "    \n",
+    "    #TODO: there's a race between the importer's background threads feeding the example queue and the training code\n",
+    "    #      calling next_batch. AFAICT this should never happen as that's precisely what TF queues are supposed to\n",
+    "    #      handle. for now, just wait a while before starting training so we can fill the example queue.\n",
+    "    time.sleep(30)\n",
```

Comment:
30 seconds seems like an awfully long time... If we have to wait (I assume due to a bug, as like you say, this is the entire point of queues...), could we not include some kind of wait/yield to do it based on when the queue actually fills with enough data for the importers?